core/os/android/adb/commands.go

```diff
@@ -287,6 +287,10 @@ func (b *binding) QueryPerfettoServiceState(ctx context.Context) (*device.Perfet
 	perfettoCapability := &device.PerfettoCapability{
 		GpuProfiling: &device.GPUProfiling{},
 	}
+	if b.Instance().GetConfiguration().GetOS().GetName() == "R" {
```

Comment:
`b.Instance().GetConfiguration().GetOS().GetAPIVersion() >= 30` and you don't need to add the TODO

Reply:
The APIversion is not yet assigned for R and it shows 29 currently. Once it has been assigned, I'll change this to use the APIversion instead.